### PR TITLE
Experiment: Vanilla pnpm

### PR DESCRIPTION
**Pull request by ryan-roemer**
This is just a base to try out installation / publishing before looking at better cached tasked execution and programmatic publishing.

- Add pnpm based install
- Do all workflows via pnpm
- Add manual publish instructions

Packages published:

- https://www.npmjs.com/package/@formidable/monorepo-tools-pnpm-dogs-core
- https://www.npmjs.com/package/@formidable/monorepo-tools-pnpm-dogs-beau
- https://www.npmjs.com/package/@formidable/monorepo-tools-pnpm-dogs-milo
- https://www.npmjs.com/package/@formidable/monorepo-tools-pnpm-dogs-rusty

Try it out!

```sh
$ pnpm -s dlx @formidable/monorepo-tools-pnpm-dogs-rusty 
Hello Rusty!
Here's what we've learned about you: Fourteen pounds of speedy, food-seeking, barkiness.
```

### CODE REVIEW

File: pnpm/package.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    "url": "git+https://github.com/FormidableLabs/monorepo-infra-experiments.git"
+  },
+  "author": "Ryan Roemer <ryan.roemer@formidable.com>",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/FormidableLabs/monorepo-infra-experiments/issues"
+  },
+  "homepage": "https://github.com/FormidableLabs/monorepo-infra-experiments#readme",
+  "scripts": {
+    "build": "pnpm -r --filter='./packages/**' run build"
```
Comment (scottrippey) [inline pnpm/package.json:18]: Locally, I was able to just run `pnpm -run -r build` and it worked the same as this; I'm assuming that it's picking up the `pnpm-workspace.yaml` config?

File: pnpm/pnpm-workspace.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+packages:
+  - 'packages/**'
```
Comment (scottrippey) [inline pnpm/pnpm-workspace.yaml:2]: Will we be able to specify dependency chains?

**Review by scottrippey — COMMENTED** [review]
Looks great! 
From this experiment, I learned: 
- `pnpm` uses hard-links for caching of `node_modules` ... I wonder how this will affect a large project?  (on this project, `npm install` was just as fast as `pnpm install`)
- `pnpm` is great for running a bunch of scripts in all directories
- I like how the workspace was defined separately

What are your key takeaways?

File: pnpm/package.json (hunk 1 above)
Comment (ryan-roemer) [reply pnpm/package.json:18]: Yeah, I think it picks it up the same way.

File: pnpm/pnpm-workspace.yaml (hunk 2 above)
Comment (ryan-roemer) [reply pnpm/pnpm-workspace.yaml:2]: In normal mode, `pnpm` figures out your real `package.json:dependencies` and links them. If you grep through our package names in the lock file you'll see `link` and such.

Comment (ryan-roemer) [reply pnpm/pnpm-workspace.yaml:2]: Oh, if you mean for task execution chains (like rebuilding, etc.), pnpm isn't the tool to take care of that. This is just the baseline from which to copy in a new directory and add in wireit, nx, etc.

**Comment by ryan-roemer** [comment]
@scottrippey Thanks for all the comments. I've found the tool easy to work with and zippy. My expectation for large projects with may repeated deps we should see a good speedup with pnpm.

As to the overall ecosystem, the pnpm folks have noticed that `npm` now has an accepted RFC to basically do what pnpm does: https://github.com/npm/rfcs/blob/main/accepted/0042-isolated-mode.md

**Pull request merged**